### PR TITLE
Add hack to fix preprocessing virtual libraries

### PR DESCRIPTION
**Pull request by rgrinberg**
This also fixes the reason support. This is a temporary hack until we have the memoization system ready to simplify this.

@jordwalke mind testing this?

### CODE REVIEW

File: src/pp_spec.mli [hunk 1]
```diff
@@ -0,0 +1,9 @@
+open! Stdune
```
Comment (diml) [inline src/pp_spec.mli]: Could you add a comment so it's easy to figure out what this module is about?

Comment (rgrinberg) [reply src/pp_spec.mli]: Done

**Comment by rgrinberg** [comment]
I added a comment about Pp_spec and also tried to factor some common code between Preprocessing and Pp_spec. The result doesn't look very convincing, so I wonder if it's worth it to factor things out for a temporary module.

**Comment by diml** [comment]
I personally fine with the result, it's always better to share this kind of things when we can. Maybe we should put `Pp_spec` inside `virtual_module.ml` for now?

**Comment by rgrinberg** [comment]
Done. It was internal to Virtual_rules anyway.

**Comment by rgrinberg** [comment]
I added another test specifically for preprocessing. I think this is good now.

**Comment by jordwalke** [comment]
This seems to work when I tested it! Thank you everyone.

**Pull request merged**